From 4207b271b56319e28cf8073b877f744088f0d15f Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Mon, 13 Aug 2018 11:53:53 +0530
Subject: [PATCH] Fix for audio getting paused on streaming audio over
 bluetooth

Audio is getting paused automatically when streaming audio
over bluetooth. Issue is seen on 1st time playback after
boot.

Fix the issue by requesting audio focus on stream start message.

Tracked-On: None

Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Aiswarya Cyriac <aiswarya.cyriac@intel.com>
---
 .../android/bluetooth/a2dpsink/A2dpSinkStreamHandler.java    | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/com/android/bluetooth/a2dpsink/A2dpSinkStreamHandler.java b/src/com/android/bluetooth/a2dpsink/A2dpSinkStreamHandler.java
index f442c49..ae913e5 100644
--- a/src/com/android/bluetooth/a2dpsink/A2dpSinkStreamHandler.java
+++ b/src/com/android/bluetooth/a2dpsink/A2dpSinkStreamHandler.java
@@ -113,7 +113,7 @@ public class A2dpSinkStreamHandler extends Handler {
             case SRC_STR_START:
                 mStreamAvailable = true;
                 // Always request audio focus if on TV.
-                if (isTvDevice()) {
+                if (isTvDevice() || isAutomotiveDevice()) {
                     if (mAudioFocus == AudioManager.AUDIOFOCUS_NONE) {
                         requestAudioFocus();
                     }
@@ -378,4 +378,7 @@ public class A2dpSinkStreamHandler extends Handler {
         return mContext.getPackageManager().hasSystemFeature(PackageManager.FEATURE_LEANBACK);
     }
 
+    private boolean isAutomotiveDevice() {
+        return mContext.getPackageManager().hasSystemFeature(PackageManager.FEATURE_AUTOMOTIVE);
+    }
 }
-- 
2.17.1

